tiny_dnn/util/product.h

```diff
@@ -516,6 +516,68 @@ void fill(T *dst, size_t size, T value) {
 
 }  // namespace detail
 
+#ifdef CNN_USE_AVX
+#ifdef CNN_USE_DOUBLE
```

Comment:
There are no usages of these classes project wide according to IntelliJ CLion so I assumed that it was some legacy code. I'll modify accordingly.